core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/include/wasmtime_ssp.h

```diff
@@ -598,6 +598,18 @@ typedef struct __wasi_addr_t {
 
 typedef enum { INET4 = 0, INET6 } __wasi_address_family_t;
 
+typedef struct __wasi_addr_info_t {
+    __wasi_addr_t addr;
+    __wasi_sock_type_t type;
+} __wasi_addr_info_t;
+
+typedef struct __wasi_addr_info_hints_t {
+   __wasi_sock_type_t type;
+   __wasi_address_family_t family;
+   // this is to workaround lack of optional parameters
+   uint8_t hints_enabled;
```

Comment:
May I ask why we need it or need it to be optional?

Reply:
I made it optional in case the user want to request all possibilities (e.g. both IPv4 or IPv6). In POSIX the way it's achieved is by specifying the `AF_UNSPEC`, but I don't think we want to add that in the `__wasi_address_family_t` enum. I'm open for suggestions though, but I think giving people ability to disable a filter is important.

Reply:
how about disabling filters when `hints` is NULL?

Reply:
Unfortunately, that's not that straightforward in WASM. When passing NULL in WASM to the function, it is translated to an address in linear memory, which no longer is NULL. We could follow the advice in [this issue](https://github.com/bytecodealliance/wasm-micro-runtime/issues/537) and use `addr_native_to_app` to see if the value passed is NULL; however, I'm still a bit concerned, as per the comment in the issue, it might not work for other languages than C. Is there already a function in WAMR that takes optional parameters so we can take it as an example? Or do you have any other suggestion for handling optional parameters?